### ISSUE TITLE
New rule: `require-slot-types`

### ISSUE THREAD

**Issue by marekdedic** [ota-meshi/eslint-plugin-svelte#347]
### Motivation

It is possible to add types to component slots and their props:

```
<script lang="ts">
    interface $$Slots {
        default: { prop: boolean; };
    }
</script>

<slot prop={true} />
```

I would like to propose a rule to enforce these slot types.

### Description

The rule should report any component that includes a `<slot>` and does not declare the `$$Slots` interface.

### Examples

```svelte
<!-- ✓ GOOD -->
<script>
</script>

<b>No slots here!</b>



<script>
  interface $$Slots {
    default: Record<string, never>;
  }
</script>

<slot />




<script lang="ts">
    interface $$Slots {
        default: { prop: boolean; };
    }
</script>

<slot prop={true} />


<script lang="ts">
    interface $$Slots {
        named: Record<string, never>;
    }
</script>

<slot name = "named" />

<!-- ✗ BAD -->
<script>
</script>

<slot />
```

### Additional comments

The relevant [RFC](https://github.com/dummdidumm/rfcs/blob/ts-typedefs-within-svelte-components/text/ts-typing-props-slots-events.md#typing-slots).

**Comment by ota-meshi** [comment]
Memo: 
New rule should be marked as experimental until the RFC is merged.
https://github.com/sveltejs/rfcs/pull/38

See also: https://github.com/ota-meshi/eslint-plugin-svelte/issues/345#issuecomment-1399695291

**Comment by marekdedic** [comment]
Relevant PR: sveltejs/language-tools#1058
